dts/bindings/gpio/infineon,tle9104.yaml

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (c) 2023 SILA Embedded Solutions GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: Infineon TLE9104 4-channel powertrain switch
+
+compatible: "infineon,tle9104"
+
+include: [gpio-controller.yaml, spi-device.yaml]
+
+properties:
+  "#gpio-cells":
+    const: 2
+
+  ngpios:
+    type: int
+    required: true
+    const: 4
+    description: Number of gpios supported
+
+  en-gpios:
+    type: phandle-array
+    description: "GPIO for enable"
+
+  resn-gpios:
+    type: phandle-array
+    required: true
+    description: "GPIO for reset"
+
+  in1-gpios:
+    type: phandle-array
+    description: "GPIO for controlling OUT1"
+
+  in2-gpios:
+    type: phandle-array
+    description: "GPIO for controlling OUT2"
+
+  in3-gpios:
+    type: phandle-array
+    description: "GPIO for controlling OUT3"
+
+  in4-gpios:
+    type: phandle-array
+    description: "GPIO for controlling OUT4"
```

Comment:
I am not sure what the purpose of these properties is?

Reply:
There are two possibilities to control the outputs, either via SPI commands or via GPIOs. Configuring one of these will ensure that the outputs are controlled via them, reducing the number of necessary SPI commands. But is not required to use them, if they are not set the outputs will be controlled via SPI commands.

Reply:
Shouldn't this just be one property, then? E.g. `in-gpios`?

Reply:
I thought about this option as well, but then the user would have to specify all the gpios or none. And it is actually possible to control the outputs only partially via dedicated GPIOs and partially via SPI. I could combine something like in-gpios with a mask property, but I am not certain if this would make the proper usage easier to understand.

Reply:
Thank you, let's leave them as separate properties, then.